diff --git a/mysql b/mysql
index dcd3345..4f454fc 100755
--- a/mysql
+++ b/mysql
@@ -110,7 +110,7 @@ case "${1:-''}" in
                 sleep 1
 	        if mysqld_status check_alive nowarn ; then break; fi
 		# wait 10sec before start checking if pid file created or server is dead
-		if [ $dead_check_counter -lt 10 ]; then
+		if [ $dead_check_counter -lt 30 ]; then
 			dead_check_counter=$(( dead_check_counter + 1 ))
 		else
 			if mysqld_status check_dead nowarn; then break; fi
